src/app/components/UserAssets/VestedAssets.tsx

```diff
@@ -1,64 +1,34 @@
-import React, { useMemo, useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Tooltip } from '@blueprintjs/core';
-import { bignumber } from 'mathjs';
 import { translations } from '../../../locales/i18n';
-import { Asset } from '../../../types';
-import { weiToFixed } from '../../../utils/blockchain/math-helpers';
-import { ethGenesisAddress } from '../../../utils/classifiers';
-import { AssetsDictionary } from '../../../utils/dictionaries/assets-dictionary';
-import {
-  numberToUSD,
-  weiToNumberFormat,
-} from '../../../utils/display-text/format';
-import { AssetDetails } from '../../../utils/models/asset-details';
-import { useCachedAssetPrice } from '../../hooks/trading/useCachedAssetPrice';
 import { useAccount, useIsConnected } from '../../hooks/useAccount';
-import { LoadableValue } from '../LoadableValue';
 import { Skeleton } from '../PageSkeleton';
-import { useVestedStaking_balanceOf } from './useVestedStaking_balanceOf';
-import { VestingDialog } from './VestingDialog';
-import { ActionButton } from 'app/components/Form/ActionButton';
-import { useMaintenance } from 'app/hooks/useMaintenance';
-import { OriginsSaleRow } from './OriginsSaleRow';
-import babelfishLogo from 'assets/images/tokens/babelfish.svg';
+import {
+  FullVesting,
+  useListOfUserVestings,
+} from './Vesting/useListOfUserVestings';
+import VestedItem from './Vesting/VestedItem';
+import { VestingWithdrawDialog } from './Vesting/VestingWithdrawDialog';
 
 export function VestedAssets() {
   const { t } = useTranslation();
   const connected = useIsConnected();
   const account = useAccount();
 
-  const item = AssetsDictionary.get(Asset.CSOV);
-  const result = useVestedStaking_balanceOf(account);
+  const { loading, items } = useListOfUserVestings();
 
   const [open, setOpen] = useState(false);
-  const [address, setAddress] = useState('');
+  const [vesting, setVesting] = useState<FullVesting>(null!);
```

Comment:
This seems a bit fishy :D If you want it to accept null, adjust it to `useState<Nullable<FullVesting>>(null)` because `!` will just tell Typescript to shut up but it's not correct.